app/assets/javascripts/admin/addon/mixins/setting-component.js

```diff
@@ -191,120 +179,37 @@ export default Mixin.create({
-  },
-
-  requiresConfirmation() {
-    return (
-      this.buffered.get("requires_confirmation") ===
-      SITE_SETTING_REQUIRES_CONFIRMATION_TYPES.simple
-    );
-  },
-
-  affectsExistingUsers() {
-    return DEFAULT_USER_PREFERENCES.includes(this.buffered.get("setting"));
-  },
-
-  confirmChanges() {
-    const settingKey = this.buffered.get("setting");
-
-    return new Promise((resolve) => {
-      // Fallback is needed in case the setting does not have a custom confirmation
-      // prompt/confirm defined.
-      this.dialog.alert({
-        message: i18n(
-          `admin.site_settings.requires_confirmation_messages.${settingKey}.prompt`,
-          {
-            translatedFallback: i18n(
-              "admin.site_settings.requires_confirmation_messages.default.prompt"
-            ),
-          }
-        ),
-        buttons: [
-          {
-            label: i18n(
-              `admin.site_settings.requires_confirmation_messages.${settingKey}.confirm`,
-              {
-                translatedFallback: i18n(
-                  "admin.site_settings.requires_confirmation_messages.default.confirm"
-                ),
-              }
-            ),
-            class: "btn-primary",
-            action: () => resolve(true),
-          },
-          {
-            label: i18n("no_value"),
-            class: "btn-default",
-            action: () => resolve(false),
-          },
-        ],
-      });
-    });
-  },
-
-  async configureBackfill() {
-    const key = this.buffered.get("setting");
-
-    const data = {
-      [key]: this.buffered.get("value"),
-    };
-
-    const result = await ajax(`/admin/site_settings/${key}/user_count.json`, {
-      type: "PUT",
-      data,
-    });
-
-    const count = result.user_count;
-
-    if (count > 0) {
-      await this.modal.show(SiteSettingDefaultCategoriesModal, {
-        model: {
-          siteSetting: { count, key: key.replaceAll("_", " ") },
-          setUpdateExistingUsers: this.setUpdateExistingUsers,
-        },
-      });
-    }
-  },
-
   update: action(async function () {
-    if (this.requiresConfirmation()) {
-      const confirm = await this.confirmChanges();
+    if (this.setting.requiresConfirmation) {
+      const confirm = await this.siteSettingChangeTracker.confirmChanges(
+        this.setting
+      );
 
       if (!confirm) {
         return;
       }
     }
 
-    if (this.affectsExistingUsers()) {
-      await this.configureBackfill();
+    if (this.setting.affectsExistingUsers) {
+      await this.siteSettingChangeTracker.configureBackfill(this.setting);
     }
 
     await this.save();
   }),
 
-  setUpdateExistingUsers: action(function (value) {
-    this.updateExistingUsers = value;
-  }),
-
   save: action(async function () {
     try {
-      this.set("isSaving", true);
+      this.setting.set("isSaving", true);
 
       await this._save();
 
-      this.set("validationMessage", null);
+      this.setting.set("validationMessage", null);
```

Comment:
Similar to isSaving, you can directly assign to validationMessage since it's a tracked property.

```suggestion
      this.setting.validationMessage = null;
```